app/src/main/java/de/westnordost/streetcomplete/map/LocationAwareMapFragment.kt

```diff
@@ -130,11 +139,36 @@ open class LocationAwareMapFragment : MapFragment() {
         locationManager.requestUpdates(2000, 1f)
     }
 
+    @SuppressLint("MissingPermission")
+    fun startPositionTrackingGPX() {
+        gpxTracking = true
+        tracks.add(ArrayList())
+        locationMapComponent?.isVisible = true
+        locationManager.requestUpdates(500, 1f)
+        tracksMapComponent?.startNewTrack(true)
+    }
+
     fun stopPositionTracking() {
         locationMapComponent?.isVisible = false
         locationManager.removeUpdates()
     }
 
+    fun stopPositionTrackingGPX() {
+        gpxTracking = false
+        tracksRecorded.clear()
+        tracks.last().forEach {
+            // time here is in milliseconds
+            // TODO: why is altitude zero in emulation?
```

Comment:
Seems general issue, see https://stackoverflow.com/questions/65325665/simulate-altitude-position-in-android-emulator